surfactant/relationships/dotnet_relationship.py

```diff
@@ -1,243 +1,229 @@
                         )
-                        probedirs.append(
-                            pathlib.PureWindowsPath(
-                                install_basepath, path, refName, refCulture
-                            ).as_posix()
+                        return False
+                    if refCulture and sw_culture and sw_culture != refCulture:
+                        logger.debug(
+                            f"[.NET][filter] skipping {sw.UUID}: culture {sw_culture} ≠ {refCulture}"
                         )
-    return probedirs
+                        return False
+            return True
+
+        # Phase 1: fs_tree lookup
+        for probe_dir in probedirs:
+            for fname in fname_variants:
+                path = normalize_path(probe_dir, fname)
+                match = sbom.get_software_by_path(path)
+                ok = bool(match and is_valid_match(match))
+                logger.debug(
+                    f"[.NET][fs_tree] {path} → {'UUID=' + match.UUID if ok else 'no match'}"
+                )
+                if ok:
+                    matched_uuids.add(match.UUID)
+                    used_method[match.UUID] = "fs_tree"
+
+        # Phase 2: Legacy installPath + fileName
+        if not matched_uuids:
+            for sw in sbom.software:
+                if not is_valid_match(sw):
+                    continue
+
+                has_ref_name = isinstance(sw.fileName, Iterable) and any(
+                    fn in (sw.fileName or []) for fn in fname_variants
+                )
+                if not has_ref_name or not isinstance(sw.installPath, Iterable):
+                    continue
+
+                for ip in sw.installPath:
+                    if any(ip.endswith(fn) for fn in fname_variants):
+                        logger.debug(f"[.NET][legacy] {refName} in {ip} → UUID={sw.UUID}")
+                        matched_uuids.add(sw.UUID)
+                        used_method[sw.UUID] = "legacy_installPath"
+
+        # Phase 3: Heuristic matching by shared directory
+        if not matched_uuids:
+            for sw in sbom.software:
+                if not is_valid_match(sw):
+                    continue
+
+                has_ref_name = isinstance(sw.fileName, Iterable) and any(
+                    fn in (sw.fileName or []) for fn in fname_variants
+                )
+                if not has_ref_name or not isinstance(sw.installPath, Iterable):
+                    continue
+
+                for sw_path in sw.installPath:
+                    sw_dir = pathlib.PurePath(sw_path).parent
+                    for refdir in probedirs:
+                        if sw_dir == pathlib.PurePath(refdir):
+                            logger.debug(
+                                f"[.NET][heuristic] {refName} via {sw_path} → UUID={sw.UUID}"
+                            )
+                            matched_uuids.add(sw.UUID)
+                            used_method[sw.UUID] = "heuristic"
```

Comment:
I think similar to some of the other instances, the shared directory heuristic here will in some edge cases add potential for false positives that didn't exist before, while the non-false positive cases look like they'd still be covered by phase 2.